lib/src/main/java/co/nimblehq/alpine/lib/mrz/MrzProcessorImpl.kt

```diff
@@ -0,0 +1,115 @@
+package co.nimblehq.alpine.lib.mrz
+
+import android.graphics.BitmapFactory
+import android.util.Log
+import co.nimblehq.alpine.lib.model.MrzInfo
+import co.nimblehq.alpine.lib.mrz.MrzProcessorException.*
+import com.google.mlkit.common.MlKitException
+import com.google.mlkit.vision.common.InputImage
+import com.google.mlkit.vision.text.*
+import com.google.mlkit.vision.text.latin.TextRecognizerOptions
+import java.io.File
+import java.util.regex.Pattern
+
+internal class MrzProcessorImpl : MrzProcessor {
+
+    companion object {
+        private const val TAG = "MlKitWrapper"
+        private const val PASSPORT_TD_3_LINE_1_REGEX = "(P[A-Z0-9<]{1})([A-Z]{3})([A-Z0-9<]{39})"
+        private const val PASSPORT_TD_3_LINE_2_REGEX =
+            "([A-Z0-9<]{9})([0-9]{1})([A-Z]{3})([0-9]{6})([0-9]{1})" +
+                "([M|F|X|<]{1})([0-9]{6})([0-9]{1})([A-Z0-9<]{14})([0-9]{1})([0-9]{1})"
+        private const val IMAGE_ORIENTATION_IN_DEGREES = 90
+        private const val DOCUMENT_NUMBER_MINIMUM_LENGTH = 8
+        private const val DATE_OF_BIRTH_MINIMUM_LENGTH = 6
+        private const val DATE_OF_EXPIRY_MINIMUM_LENGTH = 6
+    }
+
+    private val textRecognizer: TextRecognizer by lazy {
+        TextRecognition.getClient(TextRecognizerOptions.DEFAULT_OPTIONS)
+    }
+    private lateinit var mrzProcessorResultListener: MrzProcessorResultListener
+    private var scannedTextBuffer: String = ""
+    private var isMrzDetected: Boolean = false
+
+    @Throws(MlKitException::class)
```

Comment:
Do we need this annotation here if the whole method is wrapped in a try-catch? 🤔